### PR TITLE
Add builtin DS facts

### PR DESCRIPTION
**Pull request by Irma12**
https://deploy-preview-1025--roadie.netlify.app/docs/tech-insights/builtin-data-sources/

### CODE REVIEW

File: content/docs/tech-insights/builtin-data-sources/index.md [hunk 1]
```diff
@@ -10,21 +10,200 @@ description: Built-in Data Sources.
...
+| Fixed Medium Severity Alerts | Number of fixed Dependabot alerts with "medium" severity.|
+| Fixed Low Severity Alerts | Number of fixed Dependabot alerts with "low" severity.|
+| Oldest Open Alert Publish Date| Oldest alert publish date with state "open".|
+| Oldest Open Alert Update Date| Oldest alert update date with state "open".|
+
+</details>
+
+### Entity Metadata Data Source
+
+**Annotations used:**
```
Comment (Xantier) [inline content/docs/tech-insights/builtin-data-sources/index.md:64]:
```suggestion
**Annotations used:** None, applicable to entities with whatever-the-configured-entity-filter-for-this-datasource-is
```

File: content/docs/tech-insights/builtin-data-sources/index.md [hunk 2]
```diff
@@ -10,21 +10,219 @@ description: Built-in Data Sources.
...
+| Annotation Keys | The entity annotation keys.|
+| Label Keys | The entity label keys.|
+| Link Urls | Links urls associated with the entity.|
+</details>
+
+### Entity Ownership Data Source
+
+Generates facts which indicate the quality of data in the spec.owner field.
+
+**Annotations used:** None, applicable to entities which are component, domain, system, api, resource our template kind
```
Comment (Xantier) [inline content/docs/tech-insights/builtin-data-sources/index.md:99]:
```suggestion
**Annotations used:** None, applicable to entities which are component, domain, system, api, resource or template kind
```

File: content/docs/tech-insights/builtin-data-sources/index.md [hunk 3]
```diff
@@ -10,21 +10,219 @@ description: Built-in Data Sources.
...
+| Oldest Open Alert Publish Date| Oldest alert publish date with state "open".|
+| Oldest Open Alert Update Date| Oldest alert update date with state "open".|
+
+</details>
+
+### Entity Metadata Data Source
+
+Generates facts for entities you have loaded into Backstage based on the data defined in their entity YAML.
+
+**Annotations used:** None, applicable to entities which are component, domain, system, api, resource our template kind.
```
Comment (Xantier) [inline content/docs/tech-insights/builtin-data-sources/index.md:70]:
```suggestion
**Annotations used:** None, applicable to entities which are component, domain, system, api, resource or template kind.
```

File: content/docs/tech-insights/builtin-data-sources/index.md [hunk 4]
```diff
@@ -10,21 +10,219 @@ description: Built-in Data Sources.
...
+| Tags | A collection of tags for the projects of the entity. Stored as "key=value" strings.|
+| Criticality Attributes | A collection of attributes under "criticality" key for the projects of the entity.|
+| Environment Attributes | A collection of attributes under "environment" key for the projects of the entity.|
+| Lifecycle Attributes | A collection of attributes under "lifecycle" key for the projects of the entity.|
+
+</details>
+
+### TechDocs Data Source
+
+**Annotations used:** None, applicable to entities which are component, domain, system, api, resource our template kind
```
Comment (Xantier) [inline content/docs/tech-insights/builtin-data-sources/index.md:216]:
```suggestion
**Annotations used:** None, applicable to entities which are component, domain, system, api, resource or template kind
```

**Review by Xantier — APPROVED** [review]

File: content/docs/tech-insights/builtin-data-sources/index.md [hunk 5]
```diff
@@ -10,21 +10,219 @@ description: Built-in Data Sources.
 
 Built-in Data Sources use integration configurations from common Backstage plugins and annotations provided by them. You can see configuration instructions for integrations on their respective documentation pages in [the Plugins & Integrations page](/docs/integrations/).
```
Comment (dtuite) [inline content/docs/tech-insights/builtin-data-sources/index.md:11]: We should mention here somewhere that users can make their own data sources if they need something we haven't provided out of the box. We should also link to the instructions for doing that.

File: content/docs/tech-insights/builtin-data-sources/index.md [hunk 6]
```diff
@@ -10,21 +10,219 @@ description: Built-in Data Sources.
...
 
-**Annotations used by built-in data sources**
+You will find a list of built-in Data Sources, below, with annotations used by them and facts they currently include.
```
Comment (dtuite) [inline content/docs/tech-insights/builtin-data-sources/index.md:13]: There's a good chance we're going to add more facts to these data sources in future and forget to update these docs. 

Can we explain that there is also an in-app way to get the list of facts available on each data source, and that the in-app list is more likely to be accurate.

File: content/docs/tech-insights/builtin-data-sources/index.md [hunk 7]
```diff
@@ -10,21 +10,219 @@ description: Built-in Data Sources.
...
+
+### Entity Metadata Data Source
+
+Generates facts for entities you have loaded into Backstage based on the data defined in their entity YAML.
+
+**Annotations used:** None, applicable to entities which are component, domain, system, api, resource or template kind.
+ 
+<details>
+<summary> <b>Facts</b> </summary>
+| Name | Description          |
```
Comment (dtuite) [inline content/docs/tech-insights/builtin-data-sources/index.md:74]: There's something wrong with the formatting of this table. Please double check all the other tables also.

Screenshot 2023-05-20 at 12 25 46

File: content/docs/tech-insights/builtin-data-sources/index.md [hunk 8]
```diff
@@ -10,21 +10,219 @@ description: Built-in Data Sources.
...
-* **PagerDuty**
-  * `pagerduty.com/service-id`
+Imports data from Datadog which can be used to create Tech Insights checks.
 
-* **Datadog**
-  * `datadoghq.com/slo_tag`, for retrieving SLOs 
-  * `datadoghq.com/monitor_tags`, for retrieving monitors
+**Annotations used:** 
+ * `datadoghq.com/slo_tag`, for retrieving SLOs 
+ * `datadoghq.com/monitor_tags`, for retrieving monitors
```
Comment (dtuite) [inline content/docs/tech-insights/builtin-data-sources/index.md:21]: Love that we're listing the annotations used. Makes me think we should make this more obvious in the product also.

File: content/docs/tech-insights/builtin-data-sources/index.md [hunk 9]
```diff
@@ -10,21 +10,219 @@ description: Built-in Data Sources.
...
+| Name | Description          |
+| ------- | ------------------ |
+| Slo Count  | Number of SLOs configured in datadog matched based on the annotated tag.|
+| Monitor Count | Number of monitors configured in datadog with tags matching the monitor_tags annotation. |
+
+</details>
+
+### Dependabot Alerts Data Source
+
+Generates fact data from GitHub Dependabot Alerts for the entity.
```
Comment (dtuite) [inline content/docs/tech-insights/builtin-data-sources/index.md:35]: I think we need to explain how these are each authenticated. e.g. does dependabot go through the GitHub app or via a token? If we don't explain here, there's no good way for the user to know.

**Review by dtuite — CHANGES_REQUESTED** [review]

File: content/docs/tech-insights/builtin-data-sources/index.md (hunk 8 above)
Comment (Irma12) [reply content/docs/tech-insights/builtin-data-sources/index.md:21]: We do have [this one]( https://roadie.io/docs/details/required-annotations/). Do we have customers asking about this often?

File: content/docs/tech-insights/builtin-data-sources/index.md (hunk 7 above)
Comment (Irma12) [reply content/docs/tech-insights/builtin-data-sources/index.md:92]: Fixed it

File: content/docs/tech-insights/builtin-data-sources/index.md (hunk 9 above)
Comment (Irma12) [reply content/docs/tech-insights/builtin-data-sources/index.md:44]: I added Authentication part

File: content/docs/tech-insights/builtin-data-sources/index.md [hunk 10]
```diff
@@ -10,21 +10,259 @@ description: Built-in Data Sources.
 
 Built-in Data Sources use integration configurations from common Backstage plugins and annotations provided by them. You can see configuration instructions for integrations on their respective documentation pages in [the Plugins & Integrations page](/docs/integrations/).
 
-**Annotations used by built-in data sources**
+We have built several Data Sources, which you can find explained below, together with annotations used by them and facts they currently include. You can get a list of the facts available on each data source within the app as well. This information can be found on [Data Source results and details page](../data-source-details/).
```
Comment (dtuite) [inline content/docs/tech-insights/builtin-data-sources/index.md:13]:
```suggestion
We have built several Data Sources, which you can find explained below, together with annotations used by them and facts they currently include. You can get a list of the facts available on each data source within the app as well. You can learn how to access this information within the app on the [Data Source results and details page](../data-source-details/).
```

File: content/docs/tech-insights/builtin-data-sources/index.md [hunk 11]
```diff
@@ -10,21 +10,259 @@ description: Built-in Data Sources.
...
+</details>
+
+### Entity Metadata Data Source
+
+Generates facts for entities you have loaded into Backstage based on the data defined in their entity YAML.
+
+**Annotations used:** None, applicable to entities which are component, domain, system, api, resource or template kind.
+
+#### Authentication
+Authentication is done via [GitHub Apps](../../details/github-app-permissions/) or
```
Comment (dtuite) [inline content/docs/tech-insights/builtin-data-sources/index.md:86]: This one needs a bit more explanation. I actually don't understand myself. When does it use the GitHub app and when does it use the token? Does it use both?

**Review by dtuite — CHANGES_REQUESTED** [review]
Some of the text is cut off now.

Screenshot 2023-05-22 at 16 50 47

**Review by dtuite — APPROVED** [review]

**Comment by dtuite** [comment]
@Irma12 I'm going to merge this for you because it's approved and I want to mention it to some customers in the morning. Hope that's ok.

**Pull request merged**